engine/src/main/java/org/pentaho/di/repository/kdr/delegates/metastore/KettleDatabaseRepositoryMetaStore.java

```diff
@@ -0,0 +1,460 @@
+      }
+
+      return list;
+    } catch ( Exception e ) {
+      throw new MetaStoreException( "Unable to get list of element types for namespace '" + namespace + "'", e );
+    }
+
+  }
+
+  @Override
+  public List<String> getElementTypeIds( String namespace ) throws MetaStoreException {
+    List<IMetaStoreElementType> elementTypes = getElementTypes( namespace );
+    ArrayList<String> ids = new ArrayList<String>();
+    for ( IMetaStoreElementType elementType : elementTypes ) {
+      ids.add( elementType.getId() );
+    }
+    return ids;
+  }
+
+  @Override
+  public IMetaStoreElementType getElementType( String namespace, String elementTypeId ) throws MetaStoreException {
+    try {
+
+      ObjectId namespaceId = delegate.getNamespaceId( namespace );
+      if ( namespaceId == null ) {
+        return null;
+      }
+
+      RowMetaAndData elementTypeRow =
+        delegate.getElementType( new LongObjectId( new StringObjectId( elementTypeId ) ) );
+
+      return delegate.parseElementType( namespace, namespaceId, elementTypeRow );
+    } catch ( Exception e ) {
+      throw new MetaStoreException( "Unable to get element type with id '"
+        + elementTypeId + "' in namespace '" + namespace + "'", e );
+    }
+  }
+
+  @Override
+  public IMetaStoreElementType getElementTypeByName( String namespace, String elementTypeName ) throws MetaStoreException {
+    try {
+      LongObjectId namespaceId = delegate.getNamespaceId( namespace );
+      if ( namespaceId == null ) {
+        return null;
+      }
+      LongObjectId elementTypeId = delegate.getElementTypeId( namespaceId, elementTypeName );
+      if ( elementTypeId == null ) {
+        return null;
+      }
+      RowMetaAndData elementTypeRow = delegate.getElementType( elementTypeId );
+
+      return delegate.parseElementType( namespace, namespaceId, elementTypeRow );
+    } catch ( Exception e ) {
+      throw new MetaStoreException( "Unable to get element type with name '"
+        + elementTypeName + "' in namespace '" + namespace + "'", e );
+    }
+  }
+
+  @Override
+  public void updateElementType( String namespace, IMetaStoreElementType elementType ) throws MetaStoreException {
+    try {
+      ObjectId namespaceId = delegate.verifyNamespace( namespace );
+      String elementTypeId = elementType.getId();
+      if ( elementTypeId == null ) {
+        IMetaStoreElementType type = getElementTypeByName( namespace, elementType.getName() );
+        if ( type != null ) {
+          elementTypeId = type.getId();
+        }
+      }
+
+      if ( elementTypeId != null ) {
+        delegate.updateElementType(
+          namespaceId, new LongObjectId( new StringObjectId( elementType.getId() ) ), elementType );
+        repository.commit();
+      } else {
+        throw new MetaStoreException( "Unable to update element type: no id was provided and the name '"
+          + elementType.getName() + "' didn't match" );
+      }
+    } catch ( MetaStoreException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)